pkg/controller/utils/utils.go

```diff
@@ -114,6 +116,41 @@ func AddServiceWatch(c controller.Controller, name, namespace string) error {
 	})
 }
 
+func addLicenseWatch(c controller.Controller) error {
+	lic := &v3.LicenseKey{
+		TypeMeta: metav1.TypeMeta{Kind: "LicenseKey"},
+	}
+	return c.Watch(&source.Kind{Type: lic}, &handler.EnqueueRequestForObject{})
+}
+
+// WaitToAddLicenseKeyWatch will check if projectcalico.org APIs are available and if so, it will add a watch for LicenseKey
+// The completion of this operation will be signaled on a ready channel
+func WaitToAddLicenseKeyWatch(controller controller.Controller, client kubernetes.Interface, log logr.Logger, ready chan bool) {
+	maxDuration := 30 * time.Second
+	duration := 1 * time.Second
+	ticker := time.NewTicker(duration)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			duration = duration * 2
+			if duration >= maxDuration {
+				duration = maxDuration
+			}
+			ticker.Reset(duration)
+			if areCalicoAPIsReady(client) {
+				err := addLicenseWatch(controller)
+				if err != nil {
```

Comment:
I wonder if we want a way to propagate this error back to TigeraStatus? Probably not required but might be a useful enhancement.